source/ncengine/graphics2/diligent/resource/MeshBuffer.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+
+#include <span>
+
+namespace nc
+{
+namespace asset
+{
+struct MeshVertex;
+} // namespace asset
+
+namespace graphics
+{
+auto GetMeshVertexLayoutElements(uint32_t slot = 0u, uint32_t indexOffset = 0u) -> std::array<Diligent::LayoutElement, 7>;
+
+class MeshBuffer
```

Comment:
I'm leaning towards naming this the same as the other global resources, `GlobalMeshBufferResource` or adding a folder in resources called `global` and changing the other names so it looks like:
```
-resources
    -global
         -MeshBuffer
         -TextureBuffer
         -EnvironmentBuffer
```
Even though `MeshBuffer` is not bound the same as the other global resources, it makes the most sense to me to categorize as a global resource because functionally that's what it is, it is shader input that will be shared by many pipelines.

I don't think its necessary to draw the distinction (its implementation being different from the others) at the folder or filename level, because API-readability-wise, if I want to know what common shader inputs are available/predefined at the system level, I'd want them all to be grouped together and named something similar.

Reply:
If or when we get to the point of having user-defined buffers and shader reflection, I'd imagine a folder structure like this:

```
-resources
    -global
         -MeshBuffer
         -TextureBuffer
         -EnvironmentBuffer
     -user
         -ConstantBuffer
         -StructuredBuffer
```